src/location.rs

```diff
@@ -0,0 +1,101 @@
+use serde::{Deserialize, Serialize};
+use std::{fs, io};
+use std::ffi::{OsString};
+use std::fs::{DirEntry};
+use std::path::{Path, PathBuf};
+use regex::Regex;
+use cached::cached_result;
+use crate::config::Location;
+
+use super::config;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct LocationEntry {
+    pub packet: String,
+    pub time: f64,
+    pub hash: String,
+}
+
+const ID_REG: &str = "^([0-9]{8}-[0-9]{6}-[[:xdigit:]]{8})$";
+cached_result! {
+    ENTRY_CACHE: cached::UnboundCache<PathBuf, LocationEntry> = cached::UnboundCache::new();
+    fn read_entry(path: PathBuf) -> io::Result<LocationEntry> = {
+        let file = fs::File::open(path)?;
+        let entry: LocationEntry = serde_json::from_reader(file)?;
+        Ok(entry)
+    }
+}
+
+fn is_packet(name: &OsString, reg: &Regex) -> bool {
+    let o = name.to_str();
+    o.map_or(false, |s| reg.is_match(s))
+}
+
+fn get_priority(location_config: &Vec<Location>, entry: &DirEntry) -> i64 {
+    let id = entry.file_name();
+    location_config.into_iter()
+        .find(|l| OsString::from(&l.id) == id)
```

Comment:
We need OsString here because we're interacting with things passed back from the system calls about the file system?

Reply:
yeah, we're comparing against `entry.file_name()` which returns an `OsString`. could also cast the other way, should be totally innocent either way in this case